core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -766,8 +614,9 @@ public long getRawSize() {
       version = new Version(this.in);
       Magic.readAndVerify(this.in);
 
-      // Do a version check
-      if (!version.compatibleWith(BCFile.API_VERSION) && !version.equals(BCFile.API_VERSION_1)) {
+      // Do a version check - API_VERSION_2 used experimental crypto parameters, no longer supported
+      if (!version.compatibleWith(BCFile.API_VERSION_3)
+          && !version.compatibleWith(BCFile.API_VERSION_1)) {
         throw new RuntimeException("Incompatible BCFile fileBCFileVersion.");
```

Comment:
This should be a more specific exception type.

Reply:
Would be nice to include the version seen in the message.  Also if its ver 2 could give a nice message about it not being supported.

Reply:
Think throw IOException is appropriate here?  I will update the message.

Reply:
Do you need it to stay a RuntimeException? If so, there's an UncheckedIOException. Either IOException or a more specific RuntimeException would be fine with me.